### PR TITLE
Regen with latest gir/gir-files

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

File: gio/src/auto/action_map.rs [hunk 1]
```diff
@@ -54,6 +57,20 @@ pub trait ActionMapExt: IsA<ActionMap> + sealed::Sealed + 'static {
             );
         }
     }
+
+    #[cfg(feature = "v2_78")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "v2_78")))]
+    #[doc(alias = "g_action_map_remove_action_entries")]
+    fn remove_action_entries(&self, entries: &[ActionEntry]) {
```
Comment (bilelmoussaoui) [inline gio/src/auto/action_map.rs]: Should be removed or manually implemented.

File: gio/sys/Cargo.toml [hunk 2]
```diff
@@ -87,3 +88,6 @@ version = "2.74"
 
 [package.metadata.system-deps.gio_2_0.v2_76]
 version = "2.76"
+
+[package.metadata.system-deps.gio_2_0.v2_78]
+version = "2.78"
```
Comment (bilelmoussaoui) [inline gio/sys/Cargo.toml:93]: Should be lowered to 2.77

File: glib/gobject-sys/Cargo.toml [hunk 3]
```diff
@@ -71,3 +72,6 @@ version = "2.74"
 
 [package.metadata.system-deps.gobject_2_0.v2_76]
 version = "2.76"
+
+[package.metadata.system-deps.gobject_2_0.v2_78]
+version = "2.78"
```
Comment (bilelmoussaoui) [inline glib/gobject-sys/Cargo.toml:77]: same

File: glib/src/auto/functions.rs [hunk 4]
```diff
@@ -16,6 +16,33 @@ pub fn access(filename: impl AsRef<std::path::Path>, mode: i32) -> i32 {
     unsafe { ffi::g_access(filename.as_ref().to_glib_none().0, mode) }
 }
 
+#[doc(alias = "g_assertion_message_cmpint")]
+pub fn assertion_message_cmpint(
```
Comment (bilelmoussaoui) [inline glib/src/auto/functions.rs:20]: doesn't look useful to have in bindings & the ffi lacks a since annotation

File: glib/sys/src/lib.rs [hunk 5]
```diff
@@ -5715,6 +5718,17 @@ extern "C" {
         func: *const c_char,
         message: *const c_char,
     );
+    pub fn g_assertion_message_cmpint(
```
Comment (bilelmoussaoui) [inline glib/sys/src/lib.rs:5721]: needs a since annotation

File: glib/src/auto/functions.rs [hunk 6]
```diff
@@ -814,6 +841,29 @@ pub fn spawn_command_line_async(
 //    unsafe { TODO: call ffi:g_spawn_sync() }
 //}
 
+#[cfg(feature = "v2_78")]
+#[cfg_attr(docsrs, doc(cfg(feature = "v2_78")))]
+#[doc(alias = "g_timeout_add_seconds_once")]
+pub fn timeout_add_seconds_once<P: Fn() + Send + Sync + 'static>(
```
Comment (bilelmoussaoui) [inline glib/src/auto/functions.rs:847]: do we want to drop our custom implementation in glib/source.rs or make it use this directly?

Comment (sdroege) [reply glib/src/auto/functions.rs:820]: Keep it because otherwise it's only available with 2.78 :)

Comment (bilelmoussaoui) [reply glib/src/auto/functions.rs:820]: So I guess, I am going to drop the new impl.

File: gdk-pixbuf/src/auto/pixbuf.rs [hunk 7]
```diff
@@ -216,13 +216,13 @@ impl Pixbuf {
 
     #[doc(alias = "gdk_pixbuf_new_from_xpm_data")]
     #[doc(alias = "new_from_xpm_data")]
-    pub fn from_xpm_data(data: &[&str]) -> Pixbuf {
+    pub fn from_xpm_data(data: &[&str]) -> Option<Pixbuf> {
```
Comment (sdroege) [inline gdk-pixbuf/src/auto/pixbuf.rs:219]: That should probably be a `Result`. We have a gir config for that

File: gdk-pixbuf/src/auto/pixbuf.rs [hunk 8]
```diff
@@ -216,13 +216,13 @@ impl Pixbuf {
...
         unsafe { from_glib_full(ffi::gdk_pixbuf_new_from_xpm_data(data.to_glib_none().0)) }
     }
 
     #[doc(alias = "gdk_pixbuf_add_alpha")]
     #[must_use]
-    pub fn add_alpha(&self, substitute_color: bool, r: u8, g: u8, b: u8) -> Pixbuf {
+    pub fn add_alpha(&self, substitute_color: bool, r: u8, g: u8, b: u8) -> Option<Pixbuf> {
```
Comment (sdroege) [inline gdk-pixbuf/src/auto/pixbuf.rs:225]: Same

File: gdk-pixbuf/src/auto/pixbuf_format.rs [hunk 9]
```diff
@@ -59,7 +58,7 @@ impl PixbufFormat {
 
     #[doc(alias = "gdk_pixbuf_format_get_name")]
     #[doc(alias = "get_name")]
-    pub fn name(&self) -> glib::GString {
+    pub fn name(&self) -> Option<glib::GString> {
```
Comment (sdroege) [inline gdk-pixbuf/src/auto/pixbuf_format.rs:61]: These can really be None?!

File: glib/src/auto/functions.rs (hunk 4 above)
Comment (sdroege) [reply glib/src/auto/functions.rs]: Yeah all those assertion/test etc fnctions need to be manually implemented, if it all. But this is unnecessary and has very nice Rust crates for the same purpose.

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise

File: gdk-pixbuf/src/auto/pixbuf_format.rs (hunk 9 above)
Comment (sdroege) [reply gdk-pixbuf/src/auto/pixbuf_format.rs:61]: Yes

**Review by sdroege — APPROVED** [review]

**Pull request merged**